sockeye/data_io.py

```diff
@@ -415,13 +415,12 @@ def load(self,
 
         assert len(num_samples_per_bucket) == len(self.buckets)
 
-        data_source = [np.full((num_samples, source_len), self.pad_id, dtype=self.dtype)
+        num_factors = len(source_factor_sentences)
```

Comment:
if source_factor_sentences is None, len will fail. Please check whether the None use case would work in the rest of the function as well.